### PR TITLE
[Backport 2.x] Fix PSA RSA PSS verify with ALT implementations

### PR DESCRIPTION
**Pull request by mpg**
This is the 2.x backport of #4665 - almost trivial except for two differences:

- `mbedtls_rsa_set_padding()` returns `void` in 2.x instead of `int` in 3.0
- `mbedtls_rsa_rsassa_pss_verify()` still has the `mode` and RNG parameters in 2.x

### CODE REVIEW

File: include/mbedtls/rsa.h [hunk 1]
```diff
@@ -976,12 +976,13 @@ int mbedtls_rsa_rsassa_pkcs1_v15_sign( mbedtls_rsa_context *ctx,
  * \brief          This function performs a PKCS#1 v2.1 PSS signature
  *                 operation (RSASSA-PSS-SIGN).
  *
- * \note           The \p hash_id in the RSA context is the one used for the
- *                 encoding. \p md_alg in the function call is the type of hash
- *                 that is encoded. According to <em>RFC-3447: Public-Key
+ * \note           The \c hash_id set in \p ctx by calling
+ *                 mbedtls_rsa_set_padding() selects the hash used for the
```
Comment (gilles-peskine-arm) [inline include/mbedtls/rsa.h:980]: In 2.x, it can also be selected during `mbedtls_rsa_init`.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
In 2.x, the documentation needs to be different in a few places.

File: include/mbedtls/rsa.h (hunk 1 above)
Comment (mpg) [reply include/mbedtls/rsa.h]: Good point, fixed.

**Review by daverodgman — APPROVED** [review]
LGTM, if Gilles' comment above is covered.

**Review by daverodgman — COMMENTED** [review]
There seem to be a lot of places in PSA which end up calling mbedtls_rsa_set_padding, e.g. psa_verify_message. So I'm not sure this is complete - do we want to just go with something like "(when calling e.g. mbedtls_rsa_set_padding())"?

**Comment by mpg** [comment]
I think the places in PSA that end up calling `set_padding` do it on an RSA context that's internal to PSA and not supposed to be accessed directly by the user, so I don't think it's worth mentioning them here.

**Review by gilles-peskine-arm — APPROVED** [review]

**Pull request merged**